### PR TITLE
[cisco_ios] Improve parsing groks to better handle generic messages

### LINKED ISSUE #9101: [cisco_ios] Improve log parsing groks to better handle generic messages
The log parsing groks in the Cisco IOS should be reworked to better handle generic messages. Aside from the usual Cisco mnemonic messages (those beginning with a `%`), it is possible for other messages to appear, such as:
- NTP messages
- "last message repeated N times"

Currently, the first grok to run has to split the log based on the contents of the message (or at least the beginning of the message). After that, the header is then parsed. The issue with this is that in order to know where the header ends and where the message begins depends on knowing what the message looks like, which is often not the case with generic messages. There is also not a definitive character that delimits the header from the message in all cases, so the header must be parsed first.

The order of the groks should be reversed, where the header is parsed first, then the message. In practice, there are only two types of logs, Cisco logs and generic logs. Since we know how Cisco logs begin (with the mnemonic '%' character), it is trivial to separate those cases once the message portion has been separated from the header.

### PR DESCRIPTION
**Pull request by taylor-swanson**
## Proposed commit message

- The header of the log is now parsed first. A second grok runs afterwards which will look for the Cisco-mnemonic prefix for the message portion of log. If it is not found, the message will be placed into a temporary field until after Cisco messages have been processed, where it will then be placed back into the 'message' field.
- Add hit count asserts to system tests
- Add newlines to sample logs so last entries are not missed by logfile system test

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

```
cd packages/cisco_ios
elastic-package test
```

## Related issues

- Closes #9101

### CODE REVIEW

File: packages/cisco_ios/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -33,38 +33,23 @@ processors:
...
+        CISCO_PRIORITY_MSGCOUNT: '<%{NONNEGINT:log.syslog.priority:long}>(?:%{NONNEGINT:cisco.ios.message_count})?(?:: )?'
         CISCO_TIMESTAMP: '[*]?%{CISCOTIMESTAMP:_temp_.cisco_timestamp}(?: %{CISCO_TZ:_temp_.tz})?'
+        CISCO_UPTIME: '[0-9a-zA-Z]+'
+        CISCO_HOSTNAME: '[a-zA-Z][0-9a-zA-Z-]{0,61}[0-9a-zA-Z]?'
         CISCO_TZ: '[a-zA-Z]{1,4}'
-      on_failure:
-        - append:
-            field: error.message
-            value: |-
-                Processor "{{ _ingest.on_failure_processor_type }}" with tag "{{ _ingest.on_failure_processor_tag }}" in pipeline "{{ _ingest.on_failure_pipeline }}" failed with message "{{ _ingest.on_failure_message }}"
```
Comment (taylor-swanson) [inline packages/cisco_ios/data_stream/log/elasticsearch/ingest_pipeline/default.yml:62]: I'm pretty sure this is made redundant by the `on_failure` handler for the pipeline.

File: packages/cisco_ios/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -33,38 +33,23 @@ processors:
...
-        - remove:
-            field: _temp_.header
-            ignore_missing: true
-            tag: on_failure_remove_header
-
+  - grok:
+      field: _temp_.message
+      tag: grok_message
+      patterns:
+        - '^%%{GREEDYDATA:message}$'
```
Comment (gogochan) [inline packages/cisco_ios/data_stream/log/elasticsearch/ingest_pipeline/default.yml:51]: was double `%%` intentional?

Comment (taylor-swanson) [reply packages/cisco_ios/data_stream/log/elasticsearch/ingest_pipeline/default.yml:51]: Yes, `%%` is an escaped `%`. This particular processor splits Cisco mnemonic messages from generic ones, which are identified as starting with a '%':

```
%SEC-6-IPACCESSLOGNP: list ACL_CE-SECURITY denied 112 89.160.20.112 -> 224.0.0.18, 295 packets
```

versus a generic message:

```
last message repeated 66 times.
```

Full versions of those messages for context:

```
<166>352133: ASR920: Aug 3 08:04:47.140: %SEC-6-IPACCESSLOGNP: list ACL_CE-SECURITY denied 112 89.160.20.112 -> 224.0.0.18, 295 packets

<191>: rt401-rk30409: Aug 18 07:15:04.461 CEST: last message repeated 66 times.
```

**Review by gogochan — APPROVED** [review]

**Pull request merged**